### DISCUSSION TITLE
Setting title dynamically in component - How to?

### DISCUSSION THREAD

**Discussion by helmutkaufmann** [wintercms/winter#1056] [Q&A]
In my layout file for use in a CMS page, I have the following line
```
<meta name="description" content="{{ this.page.settings.meta_description | default(this.page.title) }}"/>
```
I would like to dynamically set ``this.page.settings.meta_description`` in a plugin's component I have written and basically set it to one of the parameters passed to the page the component is on (using `` input("p")``). I seem not to be able where to add that to the component's code so it gets executed before the page is rendered. I have gone through the page execution lifecycle manual pages but still... to no avail.

Any help is appreciated.

**Comment by LukeTowers** [comment]
@helmutkaufmann I'd recommend trying out the [Winter.SEO plugin](https://github.com/wintercms/wn-seo-plugin)
